### PR TITLE
Add ElfFile::CalculateDebuglinkChecksum

### PR DESCRIPTION
**Pull request by beckerhe**
Adds a function which allows to verify the CRC32 checksum as given by a
`.gnu_debuglink` section.

### CODE REVIEW

**Review by florian-kuebler — APPROVED** [review]
LGTM!

File: src/ElfUtils/ElfFile.cpp
Comment (dimitry-) [inline src/ElfUtils/ElfFile.cpp:393]: as far as I understand this file could be quite big (it most likely has .debug_info section)... maybe we will need to change to use iterative crc32 calculation instead?

Comment (dimitry-) [reply src/ElfUtils/ElfFile.cpp:393]: (add a todo?)

Comment (beckerhe) [reply src/ElfUtils/ElfFile.cpp:393]: Yes, very good point. As discussed OOB, I will add a TODO and change that to an interative approach when `orbit_base::ReadFully` is implemented in a couple of days.

Comment (beckerhe) [reply src/ElfUtils/ElfFile.cpp:395]: Added the TODO

Comment (beckerhe) [reply src/ElfUtils/ElfFile.cpp:395]: Since `ReadFully` is already implemented, I will wait until it's merged.

Comment (beckerhe) [reply src/ElfUtils/ElfFile.cpp]: @dimitry- I changed the implementation. PTAL!

File: src/ElfUtils/ElfFile.cpp
Comment (dimitry-) [inline src/ElfUtils/ElfFile.cpp:412]: chunksize_or_error could be 0 in case file size is divisible by 4MiB (not sure if StringRef handles this correctly), maybe just check for 0 and break here instead of doing the check on L415. We can afford extra read once we hit the end of file.

**Review by dimitry- — APPROVED** [review]
otherwise LGTM

File: src/ElfUtils/ElfFile.cpp
Comment (beckerhe) [reply src/ElfUtils/ElfFile.cpp:414]: done.

**Pull request merged**
